### PR TITLE
Added a SprintManager  to reorganize model dependencies between Views

### PR DESCRIPTION
**Pull request by lucas-barbero**
Changes made by @zarghol

### CODE REVIEW

File: RunningOrder/SprintManager.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//
+
+import Combine
+import SwiftUI
+
+final class SprintManager: ObservableObject {
+    @Published var sprints: [Sprint] = []
+
+    func mutableSprint(_ sprint: Sprint) -> Binding<Sprint> {
+        guard let index = sprints.firstIndex(of: sprint) else { fatalError() }
```
Comment (lucas-barbero) [inline RunningOrder/SprintManager.swift]: With this method, we have to be sure that every single sprint is unique, no check are done yet

File: RunningOrder/Story/Story.swift [hunk 2]
```diff
@@ -12,6 +12,7 @@ struct Story {
     let name: String
     let ticketReference: String
     let epic: String
+    var environment: String
```
Comment (lucas-barbero) [inline RunningOrder/Story/Story.swift]: Why did you add this var to the model ?

File: RunningOrder/SprintManager.swift (hunk 1 above)
Comment (lucas-barbero) [inline RunningOrder/SprintManager.swift]: FatalError 😠

File: RunningOrder/Story/StoryDetail.swift [hunk 3]
```diff
@@ -9,19 +9,26 @@
...
 
 struct StoryDetail: View {
-    let story: Story
+    @Binding var story: Story
 
     var body: some View {
         VStack {
             Text(story.ticketReference)
             Text(story.name)
+            Text(story.environment)
```
Comment (lucas-barbero) [inline RunningOrder/Story/StoryDetail.swift]: Do we have to keep this lines ?

File: RunningOrder/Story/StoryList.swift [hunk 4]
```diff
@@ -13,24 +13,22 @@ extension Story: Identifiable {
 }
 
 struct StoryList: View {
```
Comment (lucas-barbero) [inline RunningOrder/Story/StoryList.swift:15]: Maybe add in the futur a SprintDetail view as an intermediate like we already discussed

File: RunningOrder/SprintManager.swift (hunk 1 above)
Comment (zarghol) [reply RunningOrder/SprintManager.swift]: True ! Maybe replace fatalError with something else x) maybe error handling in case of no found. It was just quick code :)

File: RunningOrder/Story/Story.swift (hunk 2 above)
Comment (zarghol) [reply RunningOrder/Story/Story.swift]: To check something to modify and ensure everything works well :)

Comment (zarghol) [reply RunningOrder/Story/Story.swift]: You can remove it as it is not part of real model

File: RunningOrder/Story/StoryDetail.swift (hunk 3 above)
Comment (zarghol) [reply RunningOrder/Story/StoryDetail.swift]: Just for test purpose to see change applied

File: RunningOrder/Story/StoryList.swift (hunk 4 above)
Comment (zarghol) [reply RunningOrder/Story/StoryList.swift:15]: Why an intermediate ? Maybe just rebame this as SprintDetail no ?

Comment (lucas-barbero) [reply RunningOrder/Story/StoryList.swift:15]: To have a great separation of concern and perfectly match with the Master-Detail concept.
If there are a SprintDetail view and a StoryListView we'd have  
```swift
SprintDetail :
  NavigationView {
     VStack { 
          Header()
          StoryList()
    }
    Text("Select a Story")
  }
```

Comment (lucas-barbero) [reply RunningOrder/Story/StoryList.swift:15]: I think it increases readability and simplify the StoryList view which is just a List basically
